kong/plugins/aws-lambda/handler.lua

```diff
@@ -20,19 +25,64 @@ function AWSLambdaHandler:new()
 end
 
 local function retrieve_parameters()
-  ngx_req_read_body()
-
   return utils.table_merge(ngx_req_get_uri_args(), public_utils.get_body_args())
 end
 
+local function retrieve_body(content_type)
+  local raw_body = ngx_req_get_body_data()
+  if raw_body == nil then return nil end
+
+  if content_type then
+    if string_find(content_type, "application/json", nil, true) then
+      local body_json, err = cjson.decode(raw_body)
+      if not body_json then
+          ngx.log(ngx.ERR, "[aws-lambda] could not encode JSON ",
+          "body for forwarded request values: ", err)
+      end
+      return body_json
+    elseif ( string_find(content_type, "application/xml", nil, true)
+             or  string_find(content_type, "application/soap+xml", nil, true)
+             or  string_find(content_type, "text/", nil, true) )
+    then
+       return raw_body
+    end
+  end
+  return ngx_encode_base64(raw_body)
+end
+
 function AWSLambdaHandler:access(conf)
   AWSLambdaHandler.super.access(self)
 
-  local bodyJson = cjson.encode(retrieve_parameters())
+  ngx_req_read_body()
+
+  local upstream_body
+
+  if conf.forward_request_body or conf.forward_http_headers
+     or conf.forward_http_method or conf.forward_request_uri then
+     local var = ngx.var
+     upstream_body = {
+       request_body_args = not conf.forward_request_body and retrieve_parameters()
+                           or ngx_req_get_uri_args(),
```

Comment:
No If "request_body" is "true" request_body_args is going to include only uri parameters but not body itself. In case of "forward_request_body" is true body is forwarded only in request_body. The difference for Content-type json is that the body is parsed. Which repeats the existing style - parsing json for "body arguments".